wolfcrypt/src/cmac.c

```diff
@@ -79,10 +86,27 @@ int wc_InitCmac(Cmac* cmac, const byte* key, word32 keySz,
 
     (void)unused;
 
-    if (cmac == NULL || key == NULL || keySz == 0 || type != WC_CMAC_AES)
+    if (cmac == NULL || keySz == 0 || type != WC_CMAC_AES)
         return BAD_FUNC_ARG;
 
     XMEMSET(cmac, 0, sizeof(Cmac));
+
+    #ifdef WOLFSSL_QNX_CAAM
+    cmac->devId = WOLFSSL_CAAM_DEVID;
```

Comment:
Please add a `wc_InitCmac_ex` that can accept heap hint and devId. Otherwise this is not available for general use. Consider adding cmac to the crypto callback test in test.c.